src/odemis/odemisd/mdupdater.py

```diff
@@ -355,6 +355,21 @@ def observeFilter(self, filter, comp_affected):
         def updateOutWLRange(pos, fl=filter, comp_affected=comp_affected):
             wl_out = fl.axes["band"].choices[pos["band"]]
             comp_affected.updateMetadata({model.MD_OUT_WL: wl_out})
+            # apply lateral chromatic correction to align with the reference channel
+            apply_transform = fl.getMetadata().get(model.MD_CHROMATIC_COR, None)
+            if apply_transform:
+                try:
+                    metadata_cor = apply_transform[fl.position.value["band"]]
+                    assert type(metadata_cor) == dict
+                    assert all(isinstance(key, str) for key in metadata_cor)
+                    comp_affected.updateMetadata(metadata_cor)
+                except (AssertionError, KeyError) as exp:
+                    logging.error(
+                        "Chromatic correction metadata was not updated due to %s. Check if the CHROMATIC_COR is a "
+                        "dictionary with filter band positions as keys and correction metadata dictionary as "
+                        "values. For e.g. correction metadata dictionary for a given band position should "
+                        "have the following format: {\"Pixel size cor\": [1, 1],\"Centre position cor\": [0, 0]"
+                        " ,\"Rotation cor\": 0 ,\"Shear cor\": 1}", exp)
```

Comment:
I wouldn't put an example in the error message. That causes the message to be very long, and it might be repeated a lot of time in the log. Possibly, you can include an example in comment here. If someone cares of fixing the error, they should be able to find the comment.